src/main/java/io/BinaryCurveWriter.java

```diff
@@ -0,0 +1,44 @@
+package io;
+
+import algebra.curves.AbstractG1;
+import algebra.curves.AbstractG2;
+import algebra.fields.AbstractFieldElement;
+import algebra.fields.AbstractFieldElementExpanded;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.math.BigInteger;
+
+/**
+ * Base class for binary readers. Handles reading fixed-width BigInteger values (in big-endian
+ * form).
```

Comment:
This comment seems to be a residual of copy/paste from the `BinaryCurveReader`.

`reader` -> `writer`, `reading` -> `writing`

Reply:
Updated the commebt.